contract/test/dns/DNS.ts

```diff
@@ -0,0 +1,351 @@
+/* eslint-disable no-unused-expressions */
+import { expect } from 'chai'
+import { ethers, waffle } from 'hardhat'
+import { Constants, contracts, deployAll, dns } from '../utilities'
+import namehash from 'eth-ens-namehash'
+import { TestContext } from '../utilities/types'
+
+describe('DNS Tests', function () {
+  const ONE_ETH = ethers.utils.parseEther('1')
+
+  //   dns.displayNode('ETH')
+  //   dns.displayNode('country')
+  //   dns.displayNode('test.country')
+  //   dns.displayNode('test')
+
+  const TLD = process.env.TLD || 'country'
+  const DOMAIN = 'test'
+  const node = namehash.hash(DOMAIN + '.' + TLD)
+  const nameDOMAIN = (DOMAIN + '.' + TLD + '.')
+  const nameDOMAINHash = ethers.utils.keccak256(dns.dnsName(nameDOMAIN))
+  //   console.log(`nameDOMAINHash: ${nameDOMAINHash}`)
+  const aName = 'a.' + nameDOMAIN
+  const aNameHash = ethers.utils.keccak256(dns.dnsName(aName))
+  //   console.log(`aNameHash: ${aNameHash}`)
+  const bName = 'b.' + nameDOMAIN
+  const bNameHash = ethers.utils.keccak256(dns.dnsName(bName))
+  const cName = 'c.' + nameDOMAIN
+  const cNameHash = ethers.utils.keccak256(dns.dnsName(cName))
+  const dName = 'd.' + nameDOMAIN
+  const dNameHash = ethers.utils.keccak256(dns.dnsName(dName))
+  const eName = 'e.' + nameDOMAIN
+  const eNameHash = ethers.utils.keccak256(dns.dnsName(eName))
+  const fName = 'f.' + nameDOMAIN
+  const oneName = 'one.' + nameDOMAIN
+  const oneNameHash = ethers.utils.keccak256(dns.dnsName(oneName))
+
+  // Initial DNS Entries
+  // a.test.country. 3600 IN A 1.2.3.4
+  const initARec = dns.encodeARecord(aName, '1.2.3.4')
+
+  // b.test.country. 3600 IN A 2.3.4.5
+  const initB1Rec = dns.encodeARecord(bName, '2.3.4.5')
+  // b.test.country. 3600 IN A 3.4.5.6
+  const initB2Rec = dns.encodeARecord(bName, '3.4.5.6')
+  // country. 86400 IN SOA ns1.countrydns.xyz. hostmaster.test.country. 2018061501 15620 1800 1814400 14400
+  // mapping for encodeSRecord (recName, primary, admin, serial, refresh, retry, expiration, minimum) {
+  const initSOARec = dns.encodeSRecord(nameDOMAIN, 'ns1.countrydns.xyz.', 'hostmaster.test.country', '2018061501', '15620', '1800', '1814400', '14400')
+  // mapping for encodeCNAMERecord
+  const initCNAMERec = dns.encodeCNAMERecord('one.test.country', 'harmony.one')
+  const initRec = '0x' + initARec + initB1Rec + initB2Rec + initSOARec + initCNAMERec
+
+  before(async function () {
+    this.beforeSnapshotId = await waffle.provider.send('evm_snapshot', [])
+    const context = this as TestContext
```

Comment:
Use what I did in this function as an example, get rid of the use of `this` in all other functions

Reply:
should use this https://stackoverflow.com/questions/62289629/how-to-augment-types-mocha instead. I will update